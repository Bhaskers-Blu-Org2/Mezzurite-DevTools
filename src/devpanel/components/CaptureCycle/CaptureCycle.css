@media only screen {
  .capture-cycle--card {
    --box-shadow: 4px 2px 16px -2px rgba(0, 0, 0, 0.4);

    background-color: white;
    border-top: 4px solid var(--microsoft-blue);
    border-radius: 4px;
    -webkit-box-shadow: var(--box-shadow);
    -moz-box-shadow: var(--box-shadow);
    box-shadow: var(--box-shadow);
    margin: 0 16px 12px;
  }

  .capture-cycle--statistic {
    font-weight: 500;
  }

  .capture-cycle--timestamp {
    background-color: var(--microsoft-gray-lighter);
    border-bottom: 1px solid var(--microsoft-gray-darker);
    font-weight: 400;
    margin: 0;
    padding: 8px 16px;
    text-align: center;
  }
}

@media only screen and (min-width: 540px) {
  .capture-cycle--timestamp {
    text-align: start;
  }
}
