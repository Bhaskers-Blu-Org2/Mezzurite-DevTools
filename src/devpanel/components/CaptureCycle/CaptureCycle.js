import React from 'react';
import { arrayOf, number, shape, string } from 'prop-types';

import './CaptureCycle.css';
import CaptureCycleSection from '../CaptureCycleSection/CaptureCycleSection';

const CaptureCycle = (props) => {
  const captureCycleHeader = <h2 className='capture-cycle--header'>
    {props.routeUrl == null ? 'Capture Cycle' : props.routeUrl}
    {props.timestamp != null && <span> at {props.timestamp}:</span>}
  </h2>;
  const insideViewportHeading = <div className='capture-cycle--inside-viewport-header'>
    <h3 className='capture-cycle--card-header'><span className='mobile-hidden'>Components </span>Inside Viewport</h3>
    {props.viewportLoadTime != null && <h4 className='capture-cycle--viewport-load-time'>Viewport Load Time: <span className='capture-cycle--statistic'>{props.viewportLoadTime.toFixed(1)}</span>ms</h4>}
  </div>;
  const outsideViewportHeading = <h3 className='capture-cycle--card-header'><span className='mobile-hidden'>Components </span>Outside Viewport</h3>;
  const shouldRenderInsideViewportComponents = props.insideViewportComponents != null && props.insideViewportComponents.length > 0;
  const shouldRenderOutsideViewportComponents = props.outsideViewportComponents != null && props.outsideViewportComponents.length > 0;

  return (
    props != null && (shouldRenderInsideViewportComponents || shouldRenderOutsideViewportComponents) &&
    <section>
      {captureCycleHeader}
      <div className='capture-cycle--card'>
        {shouldRenderInsideViewportComponents &&
          <CaptureCycleSection
            captureCycleIndex={props.captureCycleIndex}
            components={props.insideViewportComponents}
            heading={insideViewportHeading}
          />
        }
        {shouldRenderOutsideViewportComponents &&
          <CaptureCycleSection
            captureCycleIndex={props.captureCycleIndex}
            components={props.outsideViewportComponents}
            heading={outsideViewportHeading}
          />
        }
      </div>
    </section>
  );
};

CaptureCycle.propTypes = {
  captureCycleIndex: number,
  insideViewportComponents: arrayOf(shape({
    componentLoadTime: number,
    componentName: string
  })),
  outsideViewportComponents: arrayOf(shape({
    componentLoadTime: number,
    componentName: string
  })),
  routeUrl: string,
  timestamp: string,
  viewportLoadTime: number
};

export default CaptureCycle;
