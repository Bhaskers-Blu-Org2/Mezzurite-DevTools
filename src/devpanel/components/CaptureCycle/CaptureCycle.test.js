import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import CaptureCycle from './CaptureCycle';

describe('CaptureCycle.js', () => {
  const renderer = new ShallowRenderer();

  it('should not render when the props are null', () => {
    const tree = renderer
      .render(<CaptureCycle />);
    expect(tree).toBeFalsy();
  });

  it('should not render when insideViewportComponents and outsideViewportComponents are null', () => {
    const tree = renderer
      .render(<CaptureCycle insideViewportComponents={null} outsideViewportComponents={null} />);
    expect(tree).toBeFalsy();
  });

  it('should not render when insideViewportComponents and outsideViewportComponents are empty', () => {
    const tree = renderer
      .render(<CaptureCycle insideViewportComponents={[]} outsideViewportComponents={[]} />);
    expect(tree).toBeFalsy();
  });

  it('should render when there are only insideViewportComponents', () => {
    const tree = renderer
      .render(<CaptureCycle insideViewportComponents={[
        {
          componentLoadTime: 12.2,
          componentName: 'componentName'
        }
      ]} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render when there are only outsideViewportComponents', () => {
    const tree = renderer
      .render(<CaptureCycle outsideViewportComponents={[
        {
          componentLoadTime: 12.2,
          componentName: 'componentName'
        }
      ]} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render when there are both insideViewportComponents and outsideViewportComponents', () => {
    const tree = renderer
      .render(<CaptureCycle
        insideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        outsideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
      />);
    expect(tree).toMatchSnapshot();
  });

  it('should render with a routeUrl', () => {
    const tree = renderer
      .render(<CaptureCycle
        insideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        outsideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        routeUrl='routeUrl'
      />);
    expect(tree).toMatchSnapshot();
  });

  it('should render with a timestamp', () => {
    const tree = renderer
      .render(<CaptureCycle
        insideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        outsideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        timestamp='timestamp'
      />);
    expect(tree).toMatchSnapshot();
  });

  it('should render with both a routeUrl and timestamp', () => {
    const tree = renderer
      .render(<CaptureCycle
        insideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        outsideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        routeUrl='routeUrl'
        timestamp='timestamp'
      />);
    expect(tree).toMatchSnapshot();
  });

  it('should render with a viewportLoadTime', () => {
    const tree = renderer
      .render(<CaptureCycle
        insideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        outsideViewportComponents={[
          {
            componentLoadTime: 12.2,
            componentName: 'componentName'
          }
        ]}
        viewportLoadTime={15.4}
      />);
    expect(tree).toMatchSnapshot();
  });
});
