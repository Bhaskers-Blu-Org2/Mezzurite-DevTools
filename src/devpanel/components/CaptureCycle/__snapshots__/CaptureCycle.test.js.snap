// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptureCycle.js should render when there are both insideViewportComponents and outsideViewportComponents 1`] = `
<section
  className="capture-cycle--card"
>
  <h2
    className="capture-cycle--timestamp"
  >
    Capture Cycle
  </h2>
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Inside Viewport"
    subheading={false}
  />
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Outside Viewport"
  />
</section>
`;

exports[`CaptureCycle.js should render when there are only insideViewportComponents 1`] = `
<section
  className="capture-cycle--card"
>
  <h2
    className="capture-cycle--timestamp"
  >
    Capture Cycle
  </h2>
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Inside Viewport"
    subheading={false}
  />
</section>
`;

exports[`CaptureCycle.js should render when there are only outsideViewportComponents 1`] = `
<section
  className="capture-cycle--card"
>
  <h2
    className="capture-cycle--timestamp"
  >
    Capture Cycle
  </h2>
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Outside Viewport"
  />
</section>
`;

exports[`CaptureCycle.js should render with a routeUrl 1`] = `
<section
  className="capture-cycle--card"
>
  <h2
    className="capture-cycle--timestamp"
  >
    routeUrl
  </h2>
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Inside Viewport"
    subheading={false}
  />
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Outside Viewport"
  />
</section>
`;

exports[`CaptureCycle.js should render with a timestamp 1`] = `
<section
  className="capture-cycle--card"
>
  <h2
    className="capture-cycle--timestamp"
  >
    Capture Cycle
    <span>
       at 
      timestamp
      :
    </span>
  </h2>
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Inside Viewport"
    subheading={false}
  />
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Outside Viewport"
  />
</section>
`;

exports[`CaptureCycle.js should render with a viewportLoadTime 1`] = `
<section
  className="capture-cycle--card"
>
  <h2
    className="capture-cycle--timestamp"
  >
    Capture Cycle
  </h2>
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Inside Viewport"
    subheading={
      <React.Fragment>
        Viewport Load Time: 
        <span
          className="capture-cycle--statistic"
        >
          15.4
        </span>
        ms
        <a
          aria-label="What is viewport load time?"
          className="capture-cycle--tooltip"
          href="https://github.com/Microsoft/Mezzurite#viewport-load-time-vlt"
          target="_blank"
        >
          ?
        </a>
      </React.Fragment>
    }
  />
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Outside Viewport"
  />
</section>
`;

exports[`CaptureCycle.js should render with both a routeUrl and timestamp 1`] = `
<section
  className="capture-cycle--card"
>
  <h2
    className="capture-cycle--timestamp"
  >
    routeUrl
    <span>
       at 
      timestamp
      :
    </span>
  </h2>
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Inside Viewport"
    subheading={false}
  />
  <CaptureCycleSection
    components={
      Array [
        Object {
          "componentLoadTime": 12.2,
          "componentName": "componentName",
        },
      ]
    }
    heading="Outside Viewport"
  />
</section>
`;
